### PR TITLE
Fixing the imports failures happening due to changes in OpenSearch Core.

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Fixing the imports failures happening due to changes in OpenSearch Core and removed some forbidden apis.
 
### Issues Resolved
NA
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by navneet1v** [comment]
The BWC tests rolling upgrade is only failing for Java 17. Will do more deep-dive to see what is happening.

Error Log:

```
WARN ][r.suppressed             ] [knnBwcCluster-rolling-0] path: /_cluster/health, params: {wait_for_status=yellow, wait_for_nodes=>=3}
»  org.opensearch.discovery.ClusterManagerNotDiscoveredException: null
»  	at org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction$AsyncSingleAction$2.onTimeout(TransportClusterManagerNodeAction.java:348) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  	at org.opensearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:394) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  	at org.opensearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:294) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  	at org.opensearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:707) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:747) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
»  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
»  	at java.lang.Thread.run(Thread.java:833) [?:?]
»   ↑ repeated 2 times ↑
```

**Comment by naveentatikonda** [comment]
> The BWC tests rolling upgrade is only failing for Java 17. Will do more deep-dive to see what is happening.
> 
> Error Log:
> 
> ```
> WARN ][r.suppressed             ] [knnBwcCluster-rolling-0] path: /_cluster/health, params: {wait_for_status=yellow, wait_for_nodes=>=3}
> »  org.opensearch.discovery.ClusterManagerNotDiscoveredException: null
> »  	at org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction$AsyncSingleAction$2.onTimeout(TransportClusterManagerNodeAction.java:348) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:394) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:294) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:707) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:747) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
> »  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
> »  	at java.lang.Thread.run(Thread.java:833) [?:?]
> »   ↑ repeated 2 times ↑
> ```

It is not just for Java 17. The Rolling Upgrade BWC Tests are failing for both Java 11 and Java 17. I think the OpenSearch cluster is looking for upgrade from 2.7.0 to 3.0.0. But, we are doing it from 2.6.0.

File: qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java [hunk 1]
```diff
@@ -142,7 +144,7 @@ public void testKNNModelDefault() throws Exception {
 
     // KNN Delete Model test for model in Training State
     public void testDeleteTrainingModel() throws Exception {
-        byte[] testModelBlob = "hello".getBytes();
+        byte[] testModelBlob = "hello".getBytes(StandardCharsets.UTF_8);
```
Comment (martin-gaievski) [inline qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java:147]: we do have multiple places with a method calls without Charset (e.g. https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/plugin/transport/GetModelResponseTests.java#L36), why we've changed only this one?

**Comment by naveentatikonda** [comment]
I have already created a PR to fix these changes https://github.com/opensearch-project/k-NN/pull/777

File: qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java (hunk 1 above)
Comment (navneet1v) [reply qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java:147]: This came as a forbidden api, so I fixed it.

**Comment by navneet1v** [comment]
> I have already created a PR to fix these changes #777

I didn't see it. Actually the problem is because of k-NN issue Neural Search plugin build is also failing.

**Comment by naveentatikonda** [comment]
> > I have already created a PR to fix these changes #777
> 
> I didn't see it. Actually the problem is because of k-NN issue Neural Search plugin build is also failing.

Agree. I think we can skip these failing Rolling Upgrade Tests and merge the PR to fix the failing build and unblocking neural search plugin because anyways the error is not related to these changes.

**Review by martin-gaievski — APPROVED** [review]

**Comment by navneet1v** [comment]
> The BWC tests rolling upgrade is only failing for Java 17. Will do more deep-dive to see what is happening.
> 
> Error Log:
> 
> ```
> WARN ][r.suppressed             ] [knnBwcCluster-rolling-0] path: /_cluster/health, params: {wait_for_status=yellow, wait_for_nodes=>=3}
> »  org.opensearch.discovery.ClusterManagerNotDiscoveredException: null
> »  	at org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction$AsyncSingleAction$2.onTimeout(TransportClusterManagerNodeAction.java:348) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:394) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:294) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:707) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:747) [opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
> »  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
> »  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
> »  	at java.lang.Thread.run(Thread.java:833) [?:?]
> »   ↑ repeated 2 times ↑
> ```

This will be fixed when we will have a 2.7 artifact available for k-NN and we update the Rolling upgrade tests to 2.7 version. Reason why we need 2.7 : https://github.com/opensearch-project/OpenSearch/issues/3615#issuecomment-1247544266

File: qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java (hunk 1 above)
Comment (naveentatikonda) [reply qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/ModelIT.java:147]: Ohh...I didn't run into this issue when I made the changes. Not sure if OpenSearch core made these changes after I raised the PR

**Review by naveentatikonda — APPROVED** [review]

**Pull request merged**